src/main/scala/io/github/pauljamescleary/petstore/model/OrderStatus.scala

```diff
@@ -1,20 +1,66 @@
 package io.github.pauljamescleary.petstore.model
 
 sealed trait OrderStatus extends Product with Serializable
-case object Approved extends OrderStatus
-case object Delivered extends OrderStatus
-case object Placed extends OrderStatus
 
+/*
+ * Putting case object in the object with the same name
+ * as you sealed trait is a common practice that allows
+ * - to qualify named (better discoverability for other dev)
+ *   ex: OrderStatus.Approved is self-telling, don't need to
+ *   wonder if it is not Status.Approved)
+ * - to use libraries which needs that convention
+ *   (ex: sealerate is a neat little tool that I used frequently
+ *   when I need to iterate on all cases of the ADT)
+ * - it is widelly use in scalaz/monix/shapeless/etc to better
+ *   managed implicits scoping
+ */
```

Comment:
these are really good points.  I am going to revisit the model once I wrap up some bigger pieces like authentication and scalajs.  Specifically, I would like to better enforce invariants, and your comments here are applicable with that scope of work